src/runtime_src/core/edge/skd/xrt_skd.cpp

```diff
@@ -35,313 +40,370 @@ namespace xrt {
-    syslog(LOG_INFO,"Return offset = %d\n",return_offset);
-    syslog(LOG_INFO,"Num args = %d\n",num_args);
+    m_kernel_args = xrt_core::xclbin::get_kernel_arguments(buf, prop.size, m_sk_name);
+    m_return_offset = get_return_offset(m_kernel_args);
+    std::string msg = std::string("Return offset = ") + std::to_string(m_return_offset);
+    xrt_core::message::send(severity_level::debug, "SKD", msg);
+    msg = std::string("Num args = ")  + std::to_string(m_kernel_args.size());
+    xrt_core::message::send(severity_level::debug, "SKD", msg);
     munmap(buf, prop.size);
 
     // new device handle for the current instance
-    devHdl = xclOpen(0, NULL, XCL_QUIET);
-    xrtdHdl = xrtDeviceOpenFromXcl(devHdl);
+    m_devhdl = xclOpen(0, nullptr, XCL_QUIET);
+    if (m_devhdl == nullptr) {
+	const std::string errMsg = "Cannot open XCL device handle";
+	xrt_core::message::send(severity_level::error, "SKD", errMsg);
+    }
+    m_xrtdhdl = xrtDeviceOpenFromXcl(m_devhdl);
+    if (m_devhdl == nullptr) {
+	const std::string errMsg = "Cannot open XRT device handle";
+	xrt_core::message::send(severity_level::error, "SKD", errMsg);
+    }
 
+    // Map entire PS reserve memory space
+#ifdef SKD_MAP_BIG_BO
+    m_mem_start_vaddr = xclMapBO(m_parent_devhdl,m_parent_bo_handle,true);
+    if (m_mem_start_vaddr == MAP_FAILED) {
+      std::string msg("Cannot map PS kernel Mem BO!");
+      xrt_core::message::send(severity_level::error, "SKD", msg);
+      return -EINVAL;
+    }
+    msg = std::string("host_mem_size=") + m_mem_size
+      + ", host_mem_paddr=" + m_mem_start_paddr
+      + ", host_mem_vaddr=" + std::to_string(&m_mem_start_vaddr));
+    xrt_core::message::send(severity_level::debug, "SKD", msg);
+#endif
+    
     // Check for soft kernel init function
     kernel_init_t kernel_init;
-    std::string initExtension = "_init";
-    char sk_init[PS_KERNEL_NAME_LENGTH+initExtension.size()];
-
-    snprintf(sk_init,PS_KERNEL_NAME_LENGTH+initExtension.size(),"%s%s",sk_name,initExtension.c_str());
-    kernel_init = (kernel_init_t)dlsym(sk_handle, sk_init);
-    if (!kernel_init) {
-      syslog(LOG_INFO, "kernel init function %s not found\n", sk_init);
-    } else {
-      int ret = 0;
-      ret = xrtDeviceLoadXclbinUUID(xrtdHdl,reinterpret_cast<unsigned char*>(xclbin_uuid));
-      if(ret) {
-	syslog(LOG_ERR, "Cannot load xclbin from UUID!\n");
-	return -1;
-      } else {
-	syslog(LOG_INFO, "Finished loading xclbin from UUID.\n");
-      }
-      xrtHandle = kernel_init(devHdl,reinterpret_cast<unsigned char*>(xclbin_uuid));
-      if(xrtHandle) {
-	pass_xrtHandles = true;
-	syslog(LOG_INFO, "kernel init function found! Will pass xrtHandles to soft kernel\n");
-      } else {
-	syslog(LOG_ERR, "kernel init function did not return valid xrtHandles!\n");
-	return -1;
-      }
+    std::string sk_init(std::string(m_sk_name) + std::string("_init"));
```

Comment:
m_sk_name is already a string, no need to make it a string again.
```suggestion
    auto sk_init = m_sk_name + "_init");
```